can/bus.py

```diff
@@ -420,14 +421,26 @@ def shutdown(self) -> None:
         Called to carry out any interface specific cleanup required
         in shutting down a bus.
         """
+        if self._is_shutdown:
+            LOG.debug("%s is already shut down", self.__class__)
+            return
+
         self.stop_all_periodic_tasks()
+        self._is_shutdown = True
 
     def __enter__(self):
         return self
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.shutdown()
 
+    def __del__(self) -> None:
+        if self._is_shutdown:
+            return
+
+        self.shutdown()
+        LOG.warning("%s was not properly shut down", self.__class__)
```

Comment:
I think this is easier:

```suggestion
        if not self._is_shutdown:
            # We do some best-effort cleanup if the user forgot to properly close the bus instance
            self.shutdown()
            LOG.warning("%s was not properly shut down", self.__class__)
```

Reply:
Works for me :slightly_smiling_face: